profiles/apparmor/apparmor.go

```diff
@@ -66,25 +66,29 @@ func macroExists(m string) bool {
 // InstallDefault generates a default profile in a temp directory determined by
 // os.TempDir(), then loads the profile into the kernel using 'apparmor_parser'.
 func InstallDefault(name string) error {
-	p := profileData{
-		Name: name,
-	}
+	var (
+		cmd  *exec.Cmd
+		err  error
+		pipe io.WriteCloser
+		p    profileData
+	)
+
+	p.Name = name
+	cmd = exec.Command("apparmor_parser", "-Kr")
 
-	// Install to a temporary directory.
-	f, err := ioutil.TempFile("", name)
+	pipe, err = cmd.StdinPipe()
 	if err != nil {
 		return err
 	}
-	profilePath := f.Name()
-
-	defer f.Close()
-	defer os.Remove(profilePath)
+	defer cmd.Wait()
```

Comment:
Also you should not defer it, you need to check the return value of `Wait` to see if the command finished ok.